00_Base/src/util/command.executor.ts

```diff
@@ -298,6 +318,126 @@ export class CommandExecutor {
     );
   }
 
+  public async executeReserveNow(
+    reserveNow: ReserveNow,
+    countryCode: string,
+    partyId: string,
+  ): Promise<void> {
+    // Currently, it rejects the request if the evse_uid is missing
+    // When we have the solution to handle the optional evse_uid, we should change this
+    if (!reserveNow.evse_uid) {
+      throw new BadRequestError('Missing evse_uid');
+    }
+    const evse = await this.ocpiEvseEntityRepo.getOcpiEvseByEvseUid(
+      reserveNow.evse_uid!,
+    );
+    if (!evse) {
+      throw new NotFoundError('EVSE not found');
+    }
+    const ocpiLocation = await this.ocpiLocationRepo.readByKey(
+      reserveNow.location_id,
```

Comment:
it's misleading, but the ocpi location generated database id isn't actually the "real" ocpi location id 🙈 the core location id is the one used for all ocpi-related operations, while the ocpi location database id is just there to be there.

you should be able to do this:

```
const ocpiLocation = await this.ocpiLocationRepo.getLocationByCoreLocationId(reserveNow.location_id);
```

and it should get you the same result 🤞